river/InputDevice.zig

```diff
@@ -108,7 +108,7 @@ fn handleDestroy(listener: *wl.Listener(*wlr.InputDevice), _: *wlr.InputDevice)
 
     switch (device.wlr_device.type) {
         .keyboard => {
-            const keyboard = @fieldParentPtr(Keyboard, "device", device);
+            const keyboard = @fieldParentPtr(Keyboard, "provider", @ptrCast(*Keyboard.Provider, device));
```

Comment:
TBH I habe no idea if this will work. Probably no, thanks to alignment.

See: https://github.com/ziglang/zig/issues/6611